core/commands/refs.go

```diff
@@ -169,42 +155,50 @@ Displays the hashes of all local objects.
 `,
 	},
 
-	Run: func(req cmds.Request, res cmds.Response) {
-		ctx := req.Context()
-		n, err := req.InvocContext().GetNode()
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx := req.Context
+		n, err := cmdenv.GetNode(env)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
 		// todo: make async
 		allKeys, err := n.Blockstore.AllKeysChan(ctx)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
 		out := make(chan interface{})
-		res.SetOutput((<-chan interface{})(out))
 
 		go func() {
 			defer close(out)
 
 			for k := range allKeys {
 				select {
 				case out <- &RefWrapper{Ref: k.String()}:
```

Comment:
We could drop this goroutine / channel and just call `res.Emit` directly for each key

Reply:
@magik6k Does this mean that similar cases can be refactored in this way, like(the follow code or other commands code)

https://github.com/ipfs/go-ipfs/blob/0df07e3811cbb5294dba4ccd7110002aacb45c9d/core/commands/refs.go#L110-L135

Reply:
Yes.